### ISSUE TITLE
mount 2.39.1. Which kernel version is required?

### ISSUE THREAD

**Issue by smanolache** [util-linux/util-linux#2429]
Hello,

I've just installed the debian package mount, version 2.39.1-3. I suppose it is built without --disable-libmount-mountfd-support.

My kernel version is 5.12.5.

Now I am unable to remount filesystems:

mount -o remount,exec /var
mount: /var: mount point not mounted or bad option.
       dmesg(1) may have more information after failed mount system call.

dmesg doesn't say much:

EXT4-fs (sda7): re-mounted. Opts: (null). Quota mode: none.

I've written a minimalistic program that calls the system-call mount:

int rc = mount("", argv[2], "", MS_REMOUNT, argv[1]);
if (rc != 0) ...

It works, I can remount filesystems. But I cannot with the util-linux mount executable.

As the release notes says "supports new file descriptors based mount kernel API" my intuition is that my kernel does not have this new API. In which version did it appear? I was unable to find this information. Could you please update the documentation and say that mount won't work for kernels older than version x.y.z?

Thank you,
Sorin

**Comment by karelzak** [comment]
mount(8)/libmount should be backwardly compatible (in the following example, I use a static binary from the current git tree):
```
# findmnt /mnt/test
TARGET    SOURCE    FSTYPE OPTIONS
/mnt/test /dev/sdb1 ext4   rw,noexec,relatime,seclabel,stripe=512

# ./mount.static -o remount,exec /mnt/test

# findmnt /mnt/test
TARGET    SOURCE    FSTYPE OPTIONS
/mnt/test /dev/sdb1 ext4   rw,relatime,seclabel,stripe=512

# ./mount.static -V
mount.static from util-linux 2.39.475-c0cac-dirty (libmount 2.39.475: btrfs, namespaces, idmapping, fd-based-mount, assert, debug)

# uname -a
Linux home 4.18.0-408.el8.x86_64 #1 SMP Mon Jul 18 17:42:52 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
           ^^^^^
```


Please, try `LIBMOUNT_DEBUG=all -o remount,exec /var` to get more details  (and copy & paste to this issue).

You can also try to use `LIBMOUNT_FORCE_MOUNT2=alway` to disable the new stuff if you're affected by any not yet fixed bug.

**Comment by smanolache** [comment]
[mymount.txt](https://github.com/util-linux/util-linux/files/12335654/mymount.txt)

I think I found the cause of the error on my system. I've written the small program that I've attached. When `MOUNT_ATTR_NOSYMFOLLOW` is added to the `attr_clr` flags (as `libmount` does) the system call on my machine returns `EINVAL`. Without `MOUNT_ATTR_NOSYMFOLLOW` it works fine. Apparently the type of the filesystem (mine is ext3) does not matter. I've tried it on an ext4 filesystem and I have the same behaviour (ok without nosymfollow, error with nosymfollow).

I've checked the source files of my kernel. `MOUNT_ATTR_NOSYMFOLLOW` is missing. Nosymfollow is known as a concept and used in other system calls, only `mount_setattr` does not yet understand it in 5.12.5. So my kernel is new enough to have `mount_setattr` but not new enough for `mount_setattr` to support `MOUNT_ATTR_NOSYMFOLLOW`. I guess I have to upgrade my kernel.
